### PR TITLE
Add PE waitlisted stage

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes
- We added a new  Program Engagement Stage called Waitlisted so users can track contacts waitlisted for a Program. Existing customers are strongly encouraged to add this picklist value to the Program Engagement Stage field if it suits their needs.
image

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] ~~Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))~~
- [ ] ~~Base axe-core JEST test included for any new LWC (No critical violations)~~
- [ ] ~~CRUD/FLS is enforced in Apex~~
- [ ] ~~Permission sets are updated to account for CRUD/FLS for new object metadata~~
- [ ] ~~All modified classes are unit tested (Apex) at 100% coverage~~
- [ ] UX approval or UX not necessary
- [x] PR contains draft release notes
- [x] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [x] QA
- [x] QE story level testing completed

### CODE REVIEW

**Comment by bethbrains** [comment]
**lurch: attach W-042178

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007hbjUQAQ" target="_blank">W-042178</a>

File: force-app/main/default/objects/ProgramEngagement__c/fields/Stage__c.field-meta.xml [hunk 1]
```diff
@@ -22,6 +22,11 @@
                 <default>false</default>
                 <label>Application Denied</label>
             </value>
+            <value>
+                <fullName>Waitlisted</fullName>
```
Comment (microveldt) [inline force-app/main/default/objects/ProgramEngagement__c/fields/Stage__c.field-meta.xml:26]: EDA seems to use the stage Waitlist as opposed to Waitlisted. I'm checking with Docs to see if that's their preference.

Comment (microveldt) [reply force-app/main/default/objects/ProgramEngagement__c/fields/Stage__c.field-meta.xml:26]: Docs thinks Waitlisted is most consistent with our other stages. We are good to go.

**Pull request merged**